src/test/java/io/jenkins/plugins/checks/github/GitSCMChecksContextTest.java

```diff
@@ -0,0 +1,37 @@
+package io.jenkins.plugins.checks.github;
+
+import hudson.model.Run;
+import hudson.plugins.git.GitSCM;
+import hudson.plugins.git.UserRemoteConfig;
+import org.junit.jupiter.api.Test;
+
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class GitSCMChecksContextTest {
+    @Test
+    public void shouldGetRepository() {
+        Run run = mock(Run.class);
+        GitSCM scm = mock(GitSCM.class);
+        UserRemoteConfig config = mock(UserRemoteConfig.class);
+        SCMFacade facade = mock(SCMFacade.class);
+
+        when(facade.findGitSCM(run)).thenReturn(Optional.of(scm));
+        when(facade.getUserRemoteConfig(scm)).thenReturn(config);
+
+        for (String url : new String[]{
+                "git@197.168.2.0:jenkinsci/github-checks-plugin",
+                "git@localhost:jenkinsci/github-checks-plugin",
+                "http://github.com/jenkinsci/github-checks-plugin.git",
+                "https://github.com/jenkinsci/github-checks-plugin.git"
```

Comment:
@chri7325 @almostjulian

are there any other enterprise URLs are you using?

Reply:
@XiongKezhi currently no but we may add more in my company for different sections so I'm hoping things can be checked in a robust manner.

Reply:
My url is git@<ghcompanyservername.company.local>:<repouser>/<project>.git (all those variables could change) ,  if you're regexing the servername part and using the general pattern, I should be good, but I can't hardcode to localhost or a specific ip or anything like that.

Reply:
> if you're regexing the servername part and using the general pattern, I should be good

the regex for git protocol is `git@.+:.+\\/.+` and I took the part after after ":", should be fine.
https://github.com/jenkinsci/github-checks-plugin/pull/116/files#diff-14f6b06609ffadf765bcd87e1dfe51a8efa36fa3013fd81cec41aec757a611acR100